### ISSUE TITLE
Tanh Distribution Transform

### ISSUE THREAD

**Issue by varun-intel** [pytorch/pytorch#33195]
## 🚀 Feature

Could you please add a Tanh transform to the torch.distributions.transforms module?

## Motivation

The policy network used by the Soft Actor Critic algorithm passes its output through a tanh function. In order to optimize the policy, the algorithm requires the log probabilities of the tanh output to be calculated. A tanh transform make this much simpler to implement .

## Pitch

A class torch.distributions.transforms that applies the tanh function to random variable analogously to torch.distributions.transforms.SigmoidTransform for sigmoid.

## Alternatives

It is possible to compute the log probabilities manually, but doing so is cumbersome and error prone.

## Additional context

SAC paper: https://arxiv.org/abs/1801.01290
Official SAC implementation, which uses a tanh bijector in tensorflow: https://github.com/rail-berkeley/softlearning

cc @fritzo @neerajprad @alicanb @vishwakftw

**Comment by vishwakftw** [comment]
Hi,

Could you consider a `ComposeTransform` with `Affine()`, `Sigmoid()` and `Affine()` for the time being? We already have a working version in https://github.com/pytorch/pytorch/pull/19785, but it hasn't landed yet.
